docs/dataset/advanced-usage/set-up-natural-language-search.md

```diff
@@ -0,0 +1,157 @@
+---
+icon: kolena/comparison-16
+---
+
+# :kolena-comparison-20: Natural Language Search Setup
+
+Kolena supports natural language and similar image search
+across [`Image`][kolena.asset.ImageAsset] data registered to the platform.
+Users may set up this functionality by extracting and
+uploading the corresponding search embeddings using a Kolena provided package.
+
+## Example
+
+The [`kolena`](https://github.com/kolenaIO/kolena) repository contains a runnable
+[example](https://github.com/kolenaIO/kolena/tree/trunk/examples/dataset/search_embeddings) for embeddings extraction and
+upload. This builds off the data uploaded in the
+[semantic_segmentation](https://github.com/kolenaIO/kolena/tree/trunk/examples/dataset/semantic_segmentation)
+example dataset, and is best run after this data has been uploaded to your Kolena environment.
+
+## How to Set Up Natural Language Search
+
+Uploading embeddings to Kolena can be done in three simple steps:
+
+- [**Step 1**](#step-1-install-kolena_embeddings-package): installing dependency package
+- [**Step 2**](#step-2-load-dataset-and-model): loading dataset and model to run embedding extraction
+- [**Step 3**](#step-3-load-images-for-extraction): loading images for input to extraction library
+- [**Step 4**](#step-4-extract-and-upload-embeddings): extracting and uploading search embeddings
+
+Let's take a look at each step with example code snippets.
+
+### Step 1: Install `kolena_embeddings` Package
+
+The package can be installed via `pip` or `poetry` and requires use of your kolena token which can be created
+on the [:kolena-developer-16: Developer](https://app.kolena.io/redirect/developer) page.
+
+We first [retrieve and set](../../installing-kolena.md#initialization) our `KOLENA_TOKEN` environment variable.
+This is used by the uploader for authentication against your Kolena instance.
+
+```shell
+export KOLENA_TOKEN="********"
+```
+
+=== "`pip`"
+
+    Run the following command, making sure to replace <KOLENA_TOKEN> with the token retrieved from the developer page:
+    ```shell
+    pip install --extra-index-url="https://<KOLENA_TOKEN>@gateway.kolena.cloud/repositories" kolena-embeddings
```

Comment:
Is this still the right URL, or should it be `api.kolena.io`? CC @ly-kolena

Reply:
Hmm good reminder, I copied this from the existing doc. Tried with python 3.10 and was getting error from both endpoints:

```
WARNING: The index url "https//***@gateway.kolena.cloud/repositories" seems invalid, please provide a scheme.
Looking in indexes: https://pypi.org/simple, https//***@gateway.kolena.cloud/repositories
WARNING: Location '***@gateway.kolena.cloud/repositories/kolena-embeddings/' is ignored: it is either a non-existing path or lacks a specific scheme.
ERROR: Could not find a version that satisfies the requirement kolena-embeddings (from versions: none)
ERROR: No matching distribution found for kolena-embeddings
WARNING: Location 'https//***@gateway.kolena.cloud/repositories/pip/' is ignored: it is either a non-existing path or lacks a specific scheme.


WARNING: The index url "https//***@gateway.kolena.io/repositories" seems invalid, please provide a scheme.
Looking in indexes: https://pypi.org/simple, https//***@gateway.kolena.io/repositories
WARNING: Location 'https//***@gateway.kolena.io/repositories/kolena-embeddings/' is ignored: it is either a non-existing path or lacks a specific scheme.
ERROR: Could not find a version that satisfies the requirement kolena-embeddings (from versions: none)
ERROR: No matching distribution found for kolena-embeddings
```

Removing the pip installation instructions. Our [dataset quickstart](https://docs.kolena.io/dataset/quickstart/#sdk) did not have pip either

Reply:
Both URLS worked fine for me:

```
pip install --extra-index-url="https://$KOLENA_TOKEN@gateway.kolena.cloud/repositories" kolena-embeddings
```
Or
```
pip install --extra-index-url="https://$KOLENA_TOKEN@api.kolena.io/repositories" kolena-embeddings
```

I like providing both `pip` and `poetry` options in these tutorials, so we don't force users to understand the relatively-niche `poetry` tool.

Note that this expects you to have a `KOLENA_TOKEN` in your environment.

Reply:
Ugh that was a silly miss... confirming it works for me too, adding back shortly